presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java

```diff
@@ -751,4 +766,79 @@ public long getRetainedSizeInBytes()
             return INSTANCE_SIZE + sizeOf(outputPositions);
         }
     }
+
+    private static final class MissingFieldStreamReader
+            implements SelectiveStreamReader
+    {
+        private static final int INSTANCE_SIZE = ClassLayout.parseClass(MissingFieldStreamReader.class).instanceSize();
+        private final Type outputType;
+        private int[] outputPositions;
+        private int outputPositionCount;
+
+        MissingFieldStreamReader(Type type)
+        {
+            requireNonNull(type, "type is required");
```

Comment:
combine with assignment

`this.outputType = requireNonNull(type, "type is required");`